src/App.tsx

```diff
@@ -1,24 +1,304 @@
+import { USER_ID } from './consts';
+import { Footer } from './components/Footer/Footer';
+import { ErrorMessageType } from './types/ErrorMessageType';
 
-const USER_ID = 0;
+export const App: FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessageType.NONE);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [filterType, setFilterType] = useState<FilterBy>(FilterBy.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [activeInput, setActiveInput] = useState(true);
+  const [isRequest, setIsRequest] = useState(false);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const showError = (message: ErrorMessageType) => {
+    setErrorMessage(message);
+    setTimeout(() => {
+      setErrorMessage(ErrorMessageType.NONE);
+    }, 3000);
+  };
+
+  const loadAllTodos = async () => {
+    try {
+      const todos = await getTodos(USER_ID);
+
+      setAllTodos(todos);
+      setIsRequest(true);
+    } catch {
+      showError(ErrorMessageType.LOAD);
+    } finally {
+      setTempTodo(null);
+      setActiveInput(true);
+    }
+  };
+
+  const handleRemoveTodo = useCallback(
+    async (id: number) => {
+      try {
+        setLoadingTodoIds(prev => [...prev, id]);
+        await removeTodo(id)
+          .then(() => {
+            const todosWithoutDeleted = allTodos.filter((todo) => (
+              todo.id !== id
+            ));
+
+            setAllTodos(todosWithoutDeleted);
+          });
+      } catch {
+        setErrorMessage(ErrorMessageType.DELETE);
+      } finally {
+        setLoadingTodoIds([]);
+      }
+    }, [allTodos],
+  );
+
+  useEffect(() => {
+    loadAllTodos();
+  }, []);
+
+  const handleRemoveAllCompletedTodo = useCallback(async () => {
+    const completedTodos = allTodos.filter(({ completed }) => completed);
+
+    completedTodos.forEach(async todo => {
+      try {
+        setLoadingTodoIds(prev => [...prev, todo.id]);
+        await removeTodo(todo.id)
+          .then(() => {
+            const todosWithoutDeleted = allTodos.filter(({ completed }) => (
+              !completed
+            ));
+
+            setAllTodos(todosWithoutDeleted);
+          });
+      } catch {
+        setErrorMessage(ErrorMessageType.DELETE);
+      } finally {
+        setLoadingTodoIds([]);
+      }
+    });
+  }, [allTodos]);
+
+  const visibleTodos = useMemo(() => {
+    return filterTodosByCompleted(allTodos, filterType);
+  }, [allTodos, filterType]);
+
+  const countActiveTodos = filterTodosByCompleted(
+    allTodos,
+    FilterBy.ACTIVE,
+  ).length;
+
+  const isCompletedTodo = allTodos.length - countActiveTodos > 0;
+  const hasActiveTodo = countActiveTodos > 0;
+
+  const handleAddNewTodo = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
everything that works with form should be In corresponding Form component, you should just pass callback for adding a todo to it